File: testdata/fixtures/bench/lua/vulnerable/a03_sql_injection.lua
-- Source: CWE-89 - SQL Injection via string concatenation in Lua
-- Expected: BATOU-LUA
-- OWASP: A03:2021 - Injection (SQL Injection)

local mysql = require("luasql.mysql")

local function find_user(username)
    local env = mysql.mysql()
    local conn = env:connect("app", "root", "", "localhost")
    local query = "SELECT * FROM users WHERE name = '" .. username .. "'"
    local cursor = conn:execute(query)
    return cursor
end

local input = arg[1]
find_user(input)

File: testdata/fixtures/bench/rust/vulnerable/a03_sql_injection.rs
// Source: CWE-89 - SQL Injection via format! in Rust
// Expected: BATOU-RUST
// OWASP: A03:2021 - Injection (SQL Injection)

use std::env;

fn find_user(username: &str) -> String {
    let query = format!("SELECT * FROM users WHERE name = '{}'", username);
    // db.execute(&query)
    query
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() > 1 {
        let result = find_user(&args[1]);
        println!("{}", result);
    }
}
